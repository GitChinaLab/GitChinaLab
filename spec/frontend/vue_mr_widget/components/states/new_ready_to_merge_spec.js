import { shallowMount } from '@vue/test-utils';
import ReadyToMerge from '~/vue_merge_request_widget/components/states/new_ready_to_merge.vue';

let wrapper;

function factory({ canMerge }) {
  wrapper = shallowMount(ReadyToMerge, {
    propsData: {
      mr: {},
    },
    data() {
      return { canMerge };
    },
  });
}

describe('New ready to merge state component', () => {
  afterEach(() => {
    wrapper.destroy();
  });

  it.each`
    canMerge
    ${true}
    ${false}
  `('renders permission text if canMerge ($canMerge) is false', ({ canMerge }) => {
    factory({ canMerge });

    expect(wrapper.element).toMatchSnapshot();
  });
});
