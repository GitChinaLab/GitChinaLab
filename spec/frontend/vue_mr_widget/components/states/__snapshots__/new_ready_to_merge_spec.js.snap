// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New ready to merge state component renders permission text if canMerge (false) is false 1`] = `
<div
  class="mr-widget-body media"
>
  <status-icon-stub
    status="success"
  />
   
  <p
    class="media-body gl-m-0! gl-font-weight-bold gl-text-gray-900!"
  >
    
      Ready to merge by members who can write to the target branch.
    
  </p>
</div>
`;

exports[`New ready to merge state component renders permission text if canMerge (true) is false 1`] = `
<div
  class="mr-widget-body media"
>
  <status-icon-stub
    status="success"
  />
   
  <p
    class="media-body gl-m-0! gl-font-weight-bold gl-text-gray-900!"
  >
    
      Ready to merge!
    
  </p>
</div>
`;
